Does not compile with FFmpeg 5.0 or earlier

`ch_layout` is introduced in FFmpeg 5.1:

```
../src/audiosource.cpp: In constructor 'LWAudioDecoder::LWAudioDecoder(const std::string&, int, int, const std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >&, double)':
../src/audiosource.cpp:156:36: error: 'struct AVFrame' has no member named 'ch_layout'; did you mean 'channel_layout'?
  156 |         AP.Channels = DecodeFrame->ch_layout.nb_channels;
      |                                    ^~~~~~~~~
      |                                    channel_layout
../src/audiosource.cpp:158:26: error: 'struct AVFrame' has no member named 'ch_layout'; did you mean 'channel_layout'?
  158 |         if (DecodeFrame->ch_layout.order == AV_CHANNEL_ORDER_NATIVE) {
      |                          ^~~~~~~~~
      |                          channel_layout
../src/audiosource.cpp:158:45: error: 'AV_CHANNEL_ORDER_NATIVE' was not declared in this scope
  158 |         if (DecodeFrame->ch_layout.order == AV_CHANNEL_ORDER_NATIVE) {
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
../src/audiosource.cpp:159:45: error: 'struct AVFrame' has no member named 'ch_layout'; did you mean 'channel_layout'?
  159 |             AP.ChannelLayout = DecodeFrame->ch_layout.u.mask;
      |                                             ^~~~~~~~~
      |                                             channel_layout
../src/audiosource.cpp:160:33: error: 'struct AVFrame' has no member named 'ch_layout'; did you mean 'channel_layout'?
  160 |         } else if (DecodeFrame->ch_layout.order == AV_CHANNEL_ORDER_UNSPEC) {
      |                                 ^~~~~~~~~
      |                                 channel_layout
../src/audiosource.cpp:160:52: error: 'AV_CHANNEL_ORDER_UNSPEC' was not declared in this scope
  160 |         } else if (DecodeFrame->ch_layout.order == AV_CHANNEL_ORDER_UNSPEC) {
      |                                                    ^~~~~~~~~~~~~~~~~~~~~~~
../src/audiosource.cpp:161:13: error: 'AVChannelLayout' was not declared in this scope
  161 |             AVChannelLayout ch = {};
      |             ^~~~~~~~~~~~~~~
../src/audiosource.cpp:162:40: error: 'ch' was not declared in this scope
  162 |             av_channel_layout_default(&ch, DecodeFrame->ch_layout.nb_channels);
      |                                        ^~
../src/audiosource.cpp:162:57: error: 'struct AVFrame' has no member named 'ch_layout'; did you mean 'channel_layout'?
  162 |             av_channel_layout_default(&ch, DecodeFrame->ch_layout.nb_channels);
      |                                                         ^~~~~~~~~
      |                                                         channel_layout
../src/audiosource.cpp:162:13: error: 'av_channel_layout_default' was not declared in this scope
  162 |             av_channel_layout_default(&ch, DecodeFrame->ch_layout.nb_channels);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~
../src/audiosource.cpp: In constructor 'BestAudioSource::CacheBlock::CacheBlock(int64_t, int64_t, AVFrame*)':
../src/audiosource.cpp:269:42: error: 'AVFrame' {aka 'struct AVFrame'} has no member named 'ch_layout'; did you mean 'channel_layout'?
  269 |         Storage.resize(LineSize * Frame->ch_layout.nb_channels);
      |                                          ^~~~~~~~~
      |                                          channel_layout
../src/audiosource.cpp:272:84: error: 'AVFrame' {aka 'struct AVFrame'} has no member named 'ch_layout'; did you mean 'channel_layout'?
  272 |             UnpackChannels<uint8_t>(Frame->data[0], Storage.data(), Length, Frame->ch_layout.nb_channels);
      |                                                                                    ^~~~~~~~~
      |                                                                                    channel_layout
../src/audiosource.cpp:274:85: error: 'AVFrame' {aka 'struct AVFrame'} has no member named 'ch_layout'; did you mean 'channel_layout'?
  274 |             UnpackChannels<uint16_t>(Frame->data[0], Storage.data(), Length, Frame->ch_layout.nb_channels);
      |                                                                                     ^~~~~~~~~
      |                                                                                     channel_layout
../src/audiosource.cpp:276:85: error: 'AVFrame' {aka 'struct AVFrame'} has no member named 'ch_layout'; did you mean 'channel_layout'?
  276 |             UnpackChannels<uint32_t>(Frame->data[0], Storage.data(), Length, Frame->ch_layout.nb_channels);
      |                                                                                     ^~~~~~~~~
      |                                                                                     channel_layout
../src/audiosource.cpp:278:85: error: 'AVFrame' {aka 'struct AVFrame'} has no member named 'ch_layout'; did you mean 'channel_layout'?
  278 |             UnpackChannels<uint64_t>(Frame->data[0], Storage.data(), Length, Frame->ch_layout.nb_channels);
      |                                                                                     ^~~~~~~~~
      |                                                                                     channel_layout
```

To targets FFmpeg 5.1+ only, then the version requirement should be
```
    dependency('libavcodec', version: '>=59.37.100'),
    dependency('libavformat', version: '>=59.27.100'),
```

--- a/meson.build
+++ b/meson.build
@@ -1,9 +1,11 @@
 project('BestSource', 'cpp',
-    default_options: ['buildtype=release', 'b_ndebug=if-release', 'cpp_std=c++14'],
-    meson_version: '>=0.48.0'
+    default_options: ['buildtype=release', 'b_lto=true', 'b_ndebug=if-release', 'cpp_std=c++14'],
+    license: 'MIT',
+    meson_version: '>=0.51.0',
+    version: '0.9'
 )
 
-libs = []
+link_static = get_option('link_static')
 
 sources = [
     'src/audiosource.cpp',
@@ -13,6 +15,7 @@ sources = [
     'src/vapoursynth.cpp'
 ]
 
+libs = []
 p2p_args = []
 
 if host_machine.cpu_family().startswith('x86')
@@ -26,36 +29,35 @@ libs += static_library('p2p_main',
         'libp2p/simd/cpuinfo_x86.cpp',
         'libp2p/simd/p2p_simd.cpp'
     ],
-    include_directories: ['libp2p', 'libp2p/simd'],
-    cpp_args: p2p_args
+    cpp_args: p2p_args,
+    gnu_symbol_visibility: 'hidden'
 )
 
 if host_machine.cpu_family().startswith('x86')
     p2p_args += ['-msse4.1']
 
-    libs += static_library('p2p_sse41',
-        'libp2p/simd/p2p_sse41.cpp',
-        include_directories: ['libp2p', 'libp2p/simd'],
-        cpp_args: p2p_args
+    libs += static_library('p2p_sse41', 'libp2p/simd/p2p_sse41.cpp',
+        cpp_args: p2p_args,
+        gnu_symbol_visibility: 'hidden'
     )
 endif
 
-vapoursynth_dep = dependency('vapoursynth', version: '>=R55').partial_dependency(compile_args: true, includes: true)
-jansson_dep = dependency('jansson', version: '>= 2.7', required: true)
+vapoursynth_dep = dependency('vapoursynth', version: '>=55').partial_dependency(compile_args: true, includes: true)
 
 deps = [
     vapoursynth_dep,
-    jansson_dep,
-    dependency('libavcodec', version: '>=59.37.0'),
-    dependency('libavformat', version: '>=59.27.0'),
-    dependency('libavutil', version: '>=57.28.0'),
-    dependency('libswscale', version: '>=6.7.0')
+    dependency('jansson', version: '>=2.12', static: link_static),
+    dependency('libavcodec', version: '>=59.37.0', static: link_static),
+    dependency('libavformat', version: '>=59.27.0', static: link_static),
+    dependency('libavutil', version: '>=57.28.0', static: link_static),
+    dependency('libswscale', version: '>=6.7.0', static: link_static)
 ]
 
 shared_module('bestsource', sources,
     dependencies: deps,
-    link_with: libs,
+    gnu_symbol_visibility: 'hidden',
     install: true,
-    install_dir: join_paths(vapoursynth_dep.get_pkgconfig_variable('libdir'), 'vapoursynth'),
-    gnu_symbol_visibility: 'hidden'
+    install_dir: vapoursynth_dep.get_variable(pkgconfig: 'libdir') / 'vapoursynth',
+    link_args: link_static ? '-static' : '',
+    link_with: libs
 )

--- a/meson.build
+++ b/meson.build
@@ -46,8 +46,10 @@ jansson_dep = dependency('jansson', version: '>= 2.7', required: true)
 deps = [
     vapoursynth_dep,
     jansson_dep,
-    dependency('libavcodec', version: '>=58.18.0'),
-    dependency('libavformat', version: '>=58.12.0'),
+    dependency('libavcodec', version: '>=59.37.0'),
+    dependency('libavformat', version: '>=59.27.0'),
+    dependency('libavutil', version: '>=57.28.0'),
+    dependency('libswscale', version: '>=6.7.0')
 ]
 
 shared_module('bestsource', sources,

--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,0 +1,5 @@
+option('link_static',
+    type: 'boolean',
+    value: false,
+    description: 'Use static linking'
+)

--- a/src/SrcAttribCache.cpp
+++ b/src/SrcAttribCache.cpp
@@ -36,7 +36,7 @@ extern "C" {
 }
 
 #ifdef _WIN32
-#include <shlobj_core.h>
+#include <shlobj.h>
 
 #define STAT_STRUCT_TYPE _stat64
 

--- a/src/SrcAttribCache.h
+++ b/src/SrcAttribCache.h
@@ -23,6 +23,7 @@
 
 #include <string>
 #include <map>
+#include <cstdint>
 
 struct SourceAttributes {
     struct TrackAttributes {

